### PR TITLE
feat(anta.tests): New BGP EVPN Type-2 route test

### PR DESCRIPTION
**Pull request by carl-baillargeon**
# Description
This test verifies the EVPN Type-2 routes for a given IPv4 or MAC address and VNI.

Expected Results:
    * success: If all provided VXLAN endpoints have at least one valid and active path to their EVPN Type-2 routes.
    * failure: If any of the provided VXLAN endpoints do not have at least one valid and active path to their EVPN Type-2 routes.

- Added new test
- Cleanup old BGP function
- Added new pydantic_extra_types to requirements

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have run pre-commit for code linting and typing (`pre-commit run`)
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: tests/units/anta_tests/routing/test_bgp.py [hunk 1]
```diff
@@ -1239,4 +1239,412 @@
...
+                                    "active": False,
+                                    "valid": False,
+                                },
+                            },
+                        ]
+                    },
+                },
+            },
+        ],
+        "inputs": {"vxlan_endpoints": [{"endpoint": "192.168.20.102", "vni": 10020}]},
```
Comment (MaheshGSLAB) [inline tests/units/anta_tests/routing/test_bgp.py:1545]: can we add a case where one route have two paths and at least one path is active/valid then second route also have 2 path but none of them are active/valid

File: anta/tests/routing/bgp.py [hunk 2]
```diff
@@ -447,3 +407,61 @@ def test(self) -> None:
...
+class VerifyEVPNType2Route(AntaTest):
+    """
+    This test verifies the EVPN Type-2 routes for a given IPv4 or MAC address and VNI.
+
+    Expected Results:
+        * success: If all provided VXLAN endpoints have at least one valid and active path to their EVPN Type-2 routes.
+        * failure: If any of the provided VXLAN endpoints do not have at least one valid and active path to their EVPN Type-2 routes.
+    """
+
+    name = "VerifyEVPNType2RouteIP"
```
[the remark this answers is not in the document]
Comment (carl-baillargeon) [reply anta/tests/routing/bgp.py]: Good catch. Fixed in d74d13090dc1dca2ea38122e3a5749da7c3923c5

File: tests/units/anta_tests/routing/test_bgp.py (hunk 1 above)
Comment (carl-baillargeon) [reply tests/units/anta_tests/routing/test_bgp.py]: Added in d74d13090dc1dca2ea38122e3a5749da7c3923c5

File: anta/tests/routing/bgp.py [hunk 3]
```diff
@@ -447,3 +407,61 @@ def test(self) -> None:
...
+    description = "Verifies the EVPN Type-2 routes for a given IPv4 or MAC address and VNI."
+    categories = ["routing", "bgp"]
+    commands = [AntaTemplate(template="show bgp evpn route-type mac-ip {endpoint} vni {vni}")]
+
+    class Input(AntaTest.Input):  # pylint: disable=missing-class-docstring
```
[the remark this answers is not in the document]
Comment (carl-baillargeon) [reply anta/tests/routing/bgp.py]: Fixed in bfaeeff98df82944c6228f407315a1bd4a2658a6

**Review by MaheshGSLAB — APPROVED** [review]
LGTM

**Review by titom73 — APPROVED** [review]

**Pull request merged**